code/src/main/java/com/onlinecontacttracing/storage/data-storage/NegativeUser.java

```diff
@@ -13,13 +13,20 @@
 
   @Id private String userId;
   private String userEmail;
+  private long firstLoginSeconds;
+  private long lastLoginSeconds;
 
   // Objecify requires one constructor with no parameters
   private NegativeUser() {}
   
   public NegativeUser(String id, String email) {
     userId = id;
     userEmail = email;
+    firstLoginSeconds = lastLoginSeconds = Instant.now().getEpochSecond();
+  }
+
+  public void setLastLogin() {
+    firstLoginSeconds = Instant.now().getEpochSecond();
```

Comment:
This probably is updating the wrong field. You probably want to update lastLoginInSeconds

Reply:
yeah it was copy pasted oops